boot_services/src/boot_services.rs

```diff
@@ -264,6 +264,9 @@ pub trait BootServices: Sized {
 
     fn free_pages(&self, address: usize, nb_pages: usize) -> Result<(), efi::Status>;
 
+    // Allowing Clippy's needless_lifetimes lint due to the necessity of an explicit lifetime for BootServicesBox,
+    // which is used by MemoryMap.
+    #[allow(clippy::needless_lifetimes)]

```

Comment:
This might be related to `mockall` messing with lifetime elision per https://github.com/rust-lang/rust-clippy/issues/6622.

It looks like they go away when running `cargo clippy --no-default-features`.

Given the `mockall` problem, we might want to just disable this when that feature is enabled by placing this at the top of file:

```rust
#![cfg(feature = "mockall")]
#![allow(clippy::needless_lifetimes)]
```

Reply:
I moved the suppression to be applied to the BootServices trait